packages/protocol/src/routers/ConnextHandler.sol

```diff
@@ -152,18 +171,18 @@ contract ConnextHandler {
   function executeFailedWithUpdatedArgs(
     bytes32 transferId,
     IRouter.Action[] memory actions,
-    bytes[] memory args
+    bytes[] memory args,
+    uint256 nonce
   )
     external
     onlyAllowedCaller
   {
-    FailedTxn memory txn = _failedTxns[transferId];
+    FailedTxn memory txn = _failedTxns[transferId][nonce];
     IERC20(txn.asset).approve(address(connextRouter), txn.amount);
-    try connextRouter.xBundle(txn.actions, args) {
-      delete _failedTxns[transferId];
-      emit FailedTxnExecuted(transferId, true, txn.actions, actions, txn.args, args);
+    try connextRouter.xBundle(actions, args) {
```

Comment:
Once the tx gets successfully executed, should we mark it somehow in order to prevent replaying it?

Reply:
the struct for failed txs now includes an boolean argument for "executed?". Calling an executed failed transferId will revert.